### PR TITLE
Update/1.20.6

### PR DESCRIPTION
**Pull request by Faithcaio**
**SpongeAPI** | [Sponge](https://github.com/SpongePowered/Sponge/pull/4003)

### CODE REVIEW

File: src/main/java/org/spongepowered/api/network/EngineConnectionStates.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+/**
+ * An enumeration of all the possible engine connection states.
+ */
+public final class EngineConnectionStates {
+
+    /**
+     * The supertype of all the known connection states.
+     */
+    public static final Class<EngineConnectionStates> ALL = EngineConnectionStates.class;
```
Comment (Yeregorix) [inline src/main/java/org/spongepowered/api/network/EngineConnectionStates.java]: Should be `EngineConnectionState`. Remove letter `s`

**Comment by Yeregorix** [comment]
Instead of having :
`EngineConnectionState`
`EngineConnectionIntentState`
`EngineConnectionLoginState`
`EngineConnectionConfigurationState`
`EngineConnectionGameState`
Shouldn't we have : 
`EngineConnectionState`
`EngineConnectionState.Intent`
`EngineConnectionState.Login`
`EngineConnectionState.Configuration`
`EngineConnectionState.Game`
?
Same for `ClientConnectionState` and `ServerConnectionState`.

Also why have `EngineConnectionStates` (all the constants) ?

**Comment by aromaa** [comment]
Yes, thanks.

The constants are for the `Channel`s, thats kinda how it was and I just kept the same idea.

File: src/main/java/org/spongepowered/api/network/ServerConnectionState.java [hunk 2]
```diff
@@ -27,23 +27,52 @@
...
-    @Override
-    ServerPlayer player();
+    interface Intent extends EngineConnectionState.Intent, ServerConnectionState {
+    }
 
     /**
-     * Gets the connection latency. This is constantly calculated by the server.
-     *
-     * @return The latency
+     * Represents a authenicated state between a minecraft client and server.
```
Comment (Yeregorix) [inline src/main/java/org/spongepowered/api/network/ServerConnectionState.java]: typo, `an authenticated`

File: src/main/java/org/spongepowered/api/network/ServerConnectionState.java [hunk 3]
```diff
@@ -27,23 +27,52 @@
...
-    @Override
-    ServerPlayer player();
+    interface Intent extends EngineConnectionState.Intent, ServerConnectionState {
+    }
 
     /**
-     * Gets the connection latency. This is constantly calculated by the server.
-     *
-     * @return The latency
+     * Represents an authenicated state between a minecraft client and server.
```
Comment (Yeregorix) [inline src/main/java/org/spongepowered/api/network/ServerConnectionState.java:41]: still typo, `authenicated` is missing a `t`

**Pull request merged**